src/panels/lovelace/cards/hui-button-card.js

```diff
@@ -0,0 +1,122 @@
+import { html } from '@polymer/polymer/lib/utils/html-tag.js';
+import { PolymerElement } from '@polymer/polymer/polymer-element.js';
+
+import '../../../components/ha-card.js';
+
+import toggleEntity from '../common/entity/toggle-entity.js';
+import stateIcon from '../../../common/entity/state_icon.js';
+import EventsMixin from '../../../mixins/events-mixin.js';
+
+/*
+ * @appliesMixin EventsMixin
+ */
+class HuiButtonCard extends EventsMixin(PolymerElement) {
+  static get template() {
+    return html`
+    <style>
+      ha-icon {
+        display: flex;
+        margin: auto;
+      }
+      paper-button {
+        display: flex;
+        margin: auto;
+        text-align: center;
+      }
+      .not-found {
+        flex: 1;
+        background-color: yellow;
+        padding: 8px;
+      }
+    </style>
+    <ha-card on-click='_handleClick'>
+      <paper-button>
+        <div id='Button'>
+          <ha-icon id='ButtonIcon' style='color: [[_computeColor(_stateObj)]]; width:40%; height:40%;"' icon='[[_computeIcon(_stateObj)]]'></ha-icon>
+          <span id='ButtonName'>[[_computeName(_stateObj)]]</span>
+          <span id='ButtonState'>[[_computeState(_stateObj)]]</span>
+        </div>
+      </paper-button>
+    </ha-card>
+    `;
+  }
+
+  static get properties() {
+    return {
+      hass: {
+        type: Object
+      },
+      _config: Object,
+      _stateObj: {
+        type: Object,
+        computed: '_computeStateObj(hass.states, _config.entity)'
+      },
+    };
+  }
+
+  getCardSize() {
+    return 1;
+  }
+
+  setConfig(config) {
+    if (!config || !config.entity || (config.tap_action === 'call-service' && !config.service)) throw new Error('Invalid card configuration');
+    this._config = Object.assign({ color: 'var(--primary-text-color)', show_state: true }, config);
+  }
+
+  _computeStateObj(states, entityId) {
+    return states && entityId in states ? states[entityId] : null;
+  }
+
+  _computeName(stateObj) {
+    const config = this._config;
+    if (!stateObj) {
+      this.$.Button.className = 'not-found';
+      return 'Entity not available: ' + this._config.entity;
+    }
+    return config.name ? config.name : stateObj.attributes.friendly_name;
```

Comment:
You're not resetting className again 😉 I think that you need to add an observer to stateObj and adjust these things there, or use `class$='[[_computeClassName(stateObj)]]` in the template